File: web/src/auth/roles.ts
import type { Role } from '../api/types'

export interface RoleInfo {
  role: Role
  label: string
  /** URL segment of the role's login page and home. */
  slug: 'doctor' | 'assistant' | 'reception'
  home: string
  login: string
  description: string
}

export const ROLES: Record<Role, RoleInfo> = {
  DOCTOR: {
    role: 'DOCTOR',
    label: 'Doctor',
    slug: 'doctor',
    home: '/doctor',
    login: '/login/doctor',
    description: 'Your queue, active visits and patient history.',
  },
  ASSISTANT: {
    role: 'ASSISTANT',
    label: 'Assistant',
    slug: 'assistant',
    home: '/assistant',
    login: '/login/assistant',
    description: 'Patients and queues of the doctors you assist.',
  },
  RECEPTIONIST: {
    role: 'RECEPTIONIST',
    label: 'Receptionist',
    slug: 'reception',
    home: '/reception',
    login: '/login/reception',
    description: 'Registration, appointments, check-in and payments.',
  },
}

export const ROLE_LIST: RoleInfo[] = [ROLES.DOCTOR, ROLES.ASSISTANT, ROLES.RECEPTIONIST]

export function roleBySlug(slug: string | undefined): RoleInfo | undefined {
  return ROLE_LIST.find((info) => info.slug === slug)
}

File: mobile/lib/src/auth/roles.dart
import '../models/models.dart';

/// Texts for the role-specific login screens and home dashboards.
extension RoleTexts on Role {
  String get portalDescription => switch (this) {
        Role.doctor => 'Your queue, active visits and patient history.',
        Role.assistant => 'Patients and queues of the doctors you assist.',
        Role.receptionist => 'Registration, appointments, check-in and payments.',
      };

  String get homeTitle => switch (this) {
        Role.doctor => 'My day',
        Role.assistant => 'Clinic overview',
        Role.receptionist => 'Reception',
      };

  String get waitingTitle => this == Role.doctor ? 'My queue' : 'Waiting for doctor';

  String get inVisitTitle => this == Role.doctor ? 'My active visits' : 'In visit';
}
